Add action to deploy the app to AWS

altonotch opened the pull request.

amitsh1 committed:
build and push docker, docker compose ,ecs params

amitsh1 committed:
format

amitsh1 committed:
final details

amitsh1 committed:
layer name

amitsh1 committed:
final details

amitsha2 committed:
Create main.yml

amitsh1 committed:
add env vars

amitsha2 committed:
fix env vars

amitsha2 committed:
fix by review:change install chamber

amitsha2 committed:
fix by review:change install chamber

amitsha2 committed:
fix by review

amitsha2 committed:
Merge pull request #1 from altonotch/feature/deployment_setup

Feature/deployment setup

amitsha2 committed:
Update main.yml

amitsha2 committed:
Update main.yml

amitsha2 committed:
Update main.yml

amitsha2 committed:
fix ecr_url

amitsha2 committed:
fix mistake

amitsha2 committed:
fix missing environment

amitsha2 committed:
fix deploy

amitsha2 committed:
fix docker-compose template

amitsha2 committed:
fix CMD in dockerfile.prod

amitsha2 committed:
fix dockerfile

amitsha2 committed:
fix awslogs-group

amitsha2 committed:
fix layer name

amitsha2 committed:
fix cluster name

amitsha2 committed:
longer timeout

amitsha2 committed:
fix layer name

amitsha2 committed:
fix CMD

amitsha2 committed:
fix cmd

altonotch committed:
Fix cmd in prod dockerfile

Signed-off-by: Altonotch <info@altonotch.com>

daonb commented on deployment/Dockerfile.prod:
```diff
@@ -0,0 +1,14 @@
+# base_image is define in peerbook/Dockerfile
+# to build base_image run from peerbook:
+# docker build -t base_image .
+FROM base_image
```
I don't like having two images especially as one of them can't run. Why not merge this into Dockerfile?

altonotch replied:
Base image is your initial dockerfile, which if not we needed to change in order to have a deployed version. If we merge this into your dockerfile, you and everyone else, will need to start developing differently. The other way to do this, is to utilise docker stages, when developing locally, you will need to remember to build the right stage.

daonb commented on deployment/build_and_push.sh:
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -e
+
+VERSION=$(git rev-parse --short HEAD)
```
I use SemVer, git tags and a CHANGELOG.md to track version. If you want a version, use the git tag, if you want a HASH call it a HASH.

altonotch replied:
This is the unique identifier of an image. Can call it hash, np.

daonb commented on deployment/deploy.sh:
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+VERSION="$(git rev-parse --short HEAD)"
```
HASH or` =git tag -l | tail -1`

daonb commented on deployment/docker-compose.template.yml:
```diff
@@ -0,0 +1,17 @@
+version: '3'
```
I'd appreciate a comment at the top line explaining why this file is here

altonotch replied:
This file is the way to deploy to ecs. This whole directory is the deployment scripts required to put the code in aws.

altonotch replied:
Added a comment and a link to related aws documentation.

daonb commented on deployment/ecs-params.template.yml:
```diff
@@ -0,0 +1,14 @@
+version: 1
```
ditto

altonotch replied:
Again, you need to know and understand the deployment process to ECS.

daonb replied:
the average joe just needs to know it's used in the ecs deploy shell script

daonb reviewed (changes requested):
Hi Ana,

Thanks for the clean code.  Maybe I'm missing something but I don't see anyone using `base_image` for anything. Better to keep things simpler and have just one Dockerfile.

altonotch committed:
Update deployment scripts

- rename version to hash
- add tag to image based on git tag
- add comment to explain what is docker compose/ecs params templates.

Signed-off-by: Altonotch <info@altonotch.com>

altonotch committed:
Merge branch 'tuzig:master' into master

altonotch committed:
Check if there is git tag before trying to tag an image

Signed-off-by: Altonotch <info@altonotch.com>

altonotch committed:
Fix deployed service name

Signed-off-by: Altonotch <info@altonotch.com>

altonotch committed:
Update git action - fixing get tags

Signed-off-by: Altonotch <info@altonotch.com>

altonotch committed:
Push all tags to ECR

Signed-off-by: Altonotch <info@altonotch.com>
